drivers/adc/adc_stm32.c

```diff
@@ -626,72 +721,128 @@ static void adc_stm32_unset_common_path(const struct device *dev, uint32_t PathI
 	LL_ADC_SetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc), PathInternal);
 }
 
-static void adc_stm32_teardown_channels(const struct device *dev, uint8_t channel_id)
+static void adc_stm32_teardown_channels(const struct device *dev)
```

Comment:
change that are not due to DMA, should better come in another PR

Reply:
As requested by @erwango these changes are part of a seperate commit to which it's relevant https://github.com/zephyrproject-rtos/zephyr/pull/52965/commits/3755487ba43efb083711d0d4e69853335183acf3. I would prefer to keep it as is, but if it's strictly necessary I can move it.

Reply:
If it can make the whole set being merged faster, let's split in a separate PR.
When doing so, please also provide the related sample changes so the multiple channel feature can be tested independently from DMA. This is indeed some more work to do, but a good thing overall.

Reply:
Split the PR off to #53534, meaning this PR won't build until that PR is merged. It was designed with multiple channels in mind, and I would strongly prefer not doing redudnent work getting it to work standalone.